lib/rucio/client/baseclient.py

```diff
@@ -76,7 +77,17 @@ class BaseClient:
     TOKEN_PREFIX = 'auth_token_'
     TOKEN_EXP_PREFIX = 'auth_token_exp_'
 
-    def __init__(self, rucio_host=None, auth_host=None, account=None, ca_cert=None, auth_type=None, creds=None, timeout=600, user_agent='rucio-clients', vo=None, logger=None):
+    def __init__(self,
+                 rucio_host: Optional[str] = None,
+                 auth_host: Optional[str] = None,
+                 account: Optional[str] = None,
+                 ca_cert: Optional[str] = None,
+                 auth_type: Optional[str] = None,
+                 creds: Optional[dict[str, Any]] = None,
+                 timeout: Optional[int] = 600,
+                 user_agent: Optional[str] = 'rucio-clients',
+                 vo: Optional[str] = None,
+                 logger: Callable = LOG):  # type: ignore
```

Comment:
why is this `type: ignore` needed?

Reply:
I'm not exactly sure where the problem happens, but assigning the log type to a callable (only here in the baseclient for some reason....) produces: 
`Expression of type "Logger" cannot be assigned to parameter of type "(...) -> Unknown"
  Type "Logger" cannot be assigned to type "(...) -> Unknown"`

Reply:
Instead of `Callable`, you can use `Logger` (`from logging import Logger`) and remove the `type: ignore` comment.

This is due to the default value assigned here, `LOG`, which is defined here: https://github.com/voetberg/rucio/blob/cf7717f3bfc3c9c0edb22c0c44b275c06d679c20/lib/rucio/client/baseclient.py#L49. `setup_logger` returns a `Logger` object. (offtopic: I noticed that the `LoggerFunction` we define in `types.py` is just a generic `Callable[..., Any]`; we might want to make this more specific later on)